File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03647/p03647_num0_parsed.py
# Here is one possible solution:

n, m = map(int, input().split())
edges = [tuple(map(int, input().split())) for _ in range(m)]

adj_list = [[] for _ in range(n)]
for a, b in edges:
    adj_list[a-1].append(b-1)
    adj_list[b-1].append(a-1)

for v in adj_list[0]:
    if n-1 in adj_list[v]:
        print('POSSIBLE')
        exit()

print('IMPOSSIBLE')

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03647/p03647_num0_parsed.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    int N, M;
    cin >> N >> M;

    vector<int> islands(N+1, 0);
    for (int i = 0; i < M; i++) {
        int a, b;
        cin >> a >> b;
        islands[a]++;
        islands[b]++;
    }

    for (int i = 2; i <= N - 1; i++) {
        if (islands[i] == 2) {
            cout << "POSSIBLE" << endl;
            return 0;
        }
    }

    cout << "IMPOSSIBLE" << endl;
    return 0;
}
